### PR TITLE
Load health check endpoints before others

### PR DESCRIPTION
**Pull request by dleve123**
Before this change: the profile app was redirecting
all traffic of that didn’t explicitly match another route to it’s
app. This was resulting in slow health checks that were integrating with
gravity. Considering these health check requests run every 5 seconds,
we’re putting a lot of needless traffic on gravity and other potential
downstream services.

* prettify src/lib/setup.js

Co-authored-by: Christopher Pappas <damassi.pappas@gmail.com>

----

Goal is to ensure that a system healthcheck can never cascade to other systems!

https://app.datadoghq.com/apm/resource/force/express.request/5e3bbf98774abed2?end=1544824820463&env=production&paused=false&spanID=8713293955843515405&start=1544220020463&traceID=891944206827678761

screen shot 2018-12-14 at 5 02 16 pm

### CODE REVIEW

File: src/test/acceptance/partner_profile.js [hunk 1]
```diff
@@ -63,5 +63,12 @@ describe("Partner profile page", () => {
       $.html().should.containEql("Contact")
       $.html().should.containEql("New York")
     })
+
+    it.only("should not render a page for /system/up healthcheck endpoint", async () => {
```
Comment (damassi) [inline src/test/acceptance/partner_profile.js]: Will want to remove `only`

**Comment by alloy** [comment]
Great, makes sense! 👍 

Once you’ve address @damassi’s comment, feel free to merge.

File: src/lib/setup.js [hunk 2]
```diff
@@ -240,6 +240,14 @@ export default function(app) {
     })
   }
 
+  // Routes for pinging system time and up
```
Comment (alloy) [inline src/lib/setup.js]: Would it be helpful to add a comment about the order of mounting and why this needs to be here?

Comment (dleve123) [reply src/lib/setup.js]: I was hoping that the commit message for https://github.com/artsy/force/pull/3203/commits/ba67f40462f8df69456b718462127f25acd51fa1 would be the right place to provide context. Here is it again for convenience: 

```
Load health check endpoints before others

Before this change: the profile app was redirecting
all traffic of that didn’t explicitly match another route to it’s
app. This was resulting in slow health checks that were integrating with
gravity. Considering these health check requests run every 5 seconds,
we’re putting a lot of needless traffic on gravity and other potential
downstream services.

* prettify src/lib/setup.js

Co-authored-by: Christopher Pappas <damassi.pappas@gmail.com>
``` 

Generally, I try to push context as much to commit messages as possible. Also, it seems inline with our [guidelines in artsy/README](https://github.com/artsy/README/blob/380b65bb0cffe3b8e1c45254eb3800b7016b2885/playbooks/engineer-workflow.md#commits).

Happy to move this to the source if that's the standard for this codebase though. LMK!

File: src/test/acceptance/partner_profile.js [hunk 3]
```diff
@@ -63,5 +63,12 @@ describe("Partner profile page", () => {
       $.html().should.containEql("Contact")
       $.html().should.containEql("New York")
     })
+
+    it("should not render a page for /system/up healthcheck endpoint", async () => {
```
Comment (dleve123) [inline src/test/acceptance/partner_profile.js]: @damassi and @alloy notably this test doesn't fail if we revert the changes to setup.js :( (it will fail if the system/* routes aren't present though).

@alloy: Is there a good way to write a mock expectation that in acceptance tests that a service isn't called (gravity in this case)? Chris and I weren't aware of a pattern here for Nightmare-based tests like this.

Comment (starsirius) [reply src/test/acceptance/partner_profile.js]: I've never used this but maybe [nock](https://github.com/nock/nock#expectations) is relevant? We already [included](https://github.com/artsy/force/blob/9f903d3541a1e604b6d1b1f0f3b6c58eb0712dbb/package.json#L166) nock for other purposes. Also, what if we close Gravity server for this test. Maybe it will fail for previous implementation?

Comment (dleve123) [reply src/test/acceptance/partner_profile.js]: @starsirius Thanks for the pointer, will give myself a couple of hours to see if I can figure out a nock-based approached here!

Comment (damassi) [reply src/test/acceptance/partner_profile.js]: Since this is hitting grav every 5 seconds should we go ahead and merge and then follow up with an updated test?

**Comment by dleve123** [comment]
We're working on this right now. Will merge in sometime this evening.

**Comment by dleve123** [comment]
@damassi Can you take a look at https://github.com/artsy/force/pull/3203/commits/b6601c01d0eaa84ef32e3c64b56151015469dc20 and merge if looks good. @starsirius was a wizard and figured out a legit testing approach.

**Comment by damassi** [comment]
LGTM 👍

File: src/test/acceptance/partner_profile.js (hunk 3 above)
Comment (alloy) [reply src/test/acceptance/partner_profile.js]: Alas, I have no experience with this test setup yet. Glad y'all figured it out!

File: src/lib/setup.js (hunk 2 above)
Comment (alloy) [reply src/lib/setup.js]: A commit message should indeed provide context for why the change was necessary, but the comment I’m asking for isn’t really about the change, just an explanation of the importance of order when it comes to these things. The next person that will perform a refactor in this file will likely not look through the commit history for this file.
